cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/controllers/CatsSqlAdminController.kt

```diff
@@ -102,7 +93,43 @@ class CatsSqlAdminController(
     return CleanTablesResult(tableCount = tablesDropped.size, tables = tablesDropped)
   }
 
-  private fun validateParams(currentNamespace: String?, targetNamespace: String) {
+  @PutMapping(path = ["drop_version/{version}"])
+  fun dropTablesByVersion(@PathVariable("version") dropVersion: String): CleanTablesResult {
+    validatePermissions()
+    if (!SqlSchemaVersion.values().any { it.version == dropVersion.toInt() }) {
+      throw IllegalArgumentException("Invalid CATS SQL version.")
+    }
+
+    val conn = getConnection()
+    val sql = "show tables like 'cats_v${dropVersion}_%"
+    val tablesDropped = mutableListOf<String>()
+
+    conn.use { c ->
+      val jooq = DSL.using(c, SQLDialect.MYSQL)
+      val rs = jooq.fetch(sql).intoResultSet()
+
+      while (rs.next()) {
+        val table = rs.getString(1)
+        val dropSql = "drop table `$table`"
+        log.info("Dropping $table")
+
+        jooq.query(dropSql).execute()
+        tablesDropped.add(table)
+      }
+    }
```

Comment:
Please keep persistence logic out of controllers - move this code to a Repository class (say, `SqlAdminCommandsRepository` or something) and call that instead.

Reply:
Done. I also went ahead and moved all the other persistence logic that was in this controller over to ```SqlAdminCommandsRepository```.